PowerShell Help for Microsoft.WinGet.Client (#4674)
microsoft/winget-cli: src/PowerShell/Help/Microsoft.WinGet.Client/Update-WinGetPackage.md

 [-Location <String>] [-Log <String>] [-Force] [-Header <String>] [-AllowHashMismatch]
 [-Architecture <PSProcessorArchitecture>] [-InstallerType <PSPackageInstallerType>] [-Locale <String>]
 [-Scope <PSPackageInstallScope>] [-SkipDependencies] [-Version <String>] [-Id <String>] [-Name <String>]
 [-Moniker <String>] [-Source <String>] [[-Query] <String[]>] [-MatchOption <PSPackageFieldMatchOption>]
 [-ProgressAction <ActionPreference>] [-WhatIf] [-Confirm] [<CommonParameters>]
```

### GivenSet
```
Update-WinGetPackage [-IncludeUnknown] [-Mode <PSPackageInstallMode>] [-Override <String>] [-Custom <String>]
 [-Location <String>] [-Log <String>] [-Force] [-Header <String>] [-AllowHashMismatch]
 [-Architecture <PSProcessorArchitecture>] [-InstallerType <PSPackageInstallerType>] [-Locale <String>]
 [-Scope <PSPackageInstallScope>] [-SkipDependencies] [[-PSCatalogPackage] <PSCatalogPackage>]
 [-Version <String>] [-ProgressAction <ActionPreference>] [-WhatIf] [-Confirm] [<CommonParameters>]
```

## DESCRIPTION
Updates the selected package found by searching the installed packages list.

> ## EXAMPLES

Reviewer: Given that there's no direct equivalent of `winget upgrade --all`  in `Update-WinGetPackage`, would it make  sense to add an example here using `Get-WinGetPackage | Where-Object {($_.IsUpdateAvailable -eq $true)} | Update-WinGetPackage` for achieving the same result?

Reviewer 2: Totally, useful examples are always welcome in help docs. Sometimes one may specifically look for examples with `Get-Help <cmdlet> -Examples` too.

I'd propose the more modern (PowerShell 3+) syntax of `Get-WinGetPackage | Where-Object IsUpdateAvailable | Update-WinGetPackage` because it's less busy, but other than that no objections.

```yaml
Type: System.Management.Automation.SwitchParameter
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Architecture

Specify the installer architecture.

```yaml
Type: Microsoft.WinGet.Client.PSObjects.PSProcessorArchitecture
Parameter Sets: (All)
Aliases:
Accepted values: Default, X86, Arm, X64, Arm64

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Custom

Add additional arguments to the installer command.

```yaml
Type: System.String
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Force

Force the update to run the installer.

```yaml
Type: System.Management.Automation.SwitchParameter
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Header

> Specify a custom value in the HTTP header to the REST source.

Reviewer 3: This is a string value. The header argument should be accompanied with a source argument that specifies which source the custom header should be sent to. The source specified must be a REST source type.

The client will send the header argument as a value of HTTP header named `Windows-Package-Manager`.

> Specify a custom value in the HTTP header to the REST source.

Reviewer 4: So, the resulting HTTP header would be something like `@{ Windows-Package-Manager = $Header }`?

Why would someone need to do this?

> Specify a custom value in the HTTP header to the REST source.

Reviewer 2: As a winget user, I haven't utilized this feature yet but I was thinking of using that header value on the server side to know what packages a client should get access to, just like WSUS Client-side Targeting. E.g. the winget client would identify itself as `Windows-Package-Manager = Contoso.StaffWorkstation` and the REST source restricts the available packages based on that "group name".

I don't know if that is the intended use through.

```yaml
Type: System.Management.Automation.SwitchParameter
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -InstallerType

Specify the installer type to use.

```yaml
Type: Microsoft.WinGet.Client.PSObjects.PSPackageInstallerType
Parameter Sets: (All)
Aliases:
Accepted values: Default, Inno, Wix, Msi, Nullsoft, Zip, Msix, Exe, Burn, MSStore, Portable

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Locale

Specify the installer locale.

```yaml
Type: System.String
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Location

Specify the location to install the package.

> **Note: Not all installers support this property.**

```yaml
Type: System.String
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Log

> Specify the location for the installer log.

Reviewer 3: This is a file path that is passed as an argument to the installer. The file name can be a custom one like `foo.log` or `bar.txt`. This is only supported if the package manifest contains a log `installerSwitch` like [this example](https://github.com/microsoft/winget-pkgs/blob/ec910ce2fe9f3245c21a4383340b5b24c9cd898f/manifests/b/Brave/Brave/116.1.57.49/Brave.Brave.installer.yaml#L10).

> Specify the location for the installer log.

Reviewer 4: ```suggestion
Specify the location for the installer log. The value can be a fully-qualified or relative path and
must include the file name. For example: `$env:TEMP\package.log`.
```

Accept wildcard characters: False
```

### -Mode

Specify the installer output mode. Options are "Default", "Silent", and "Interactive".

```yaml
Type: Microsoft.WinGet.Client.PSObjects.PSPackageInstallMode
Parameter Sets: (All)
Aliases:
Accepted values: Default, Silent, Interactive

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Moniker

Specify the moniker.

```yaml
Type: System.String
Parameter Sets: FoundSet
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Name

Specify the name.

```yaml
Type: System.String
Parameter Sets: FoundSet
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Override

Override the switches to pass to the installer.

```yaml
Type: System.String
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -PSCatalogPackage

> WinGet package object

Reviewer 4: ```suggestion
Provide **PSCatalogPackage** object. You can get a **PSCatalogPackage** object by using the
`Find-WinGetPackage` or `Get-WinGetPackage` commands.
```

```yaml
Type: Microsoft.WinGet.Client.Engine.PSObjects.PSCatalogPackage
Parameter Sets: GivenSet
Aliases: InputObject

Required: False
Position: 0
Default value: None
Accept pipeline input: True (ByPropertyName, ByValue)
Accept wildcard characters: False
```

### -Query

> Search string. Wildcards are not supported.

Reviewer 4: ```suggestion
Specify one or more strings to search for. By default, the command searches all configured sources.
Wildcards are not supported. The command compares the value provided to the following package
manifest properties:

	- `PackageIdentifier`
	- `PackageName`
	- `Moniker`
	- `Tags`

The command does a case-insensitive substring comparison of these properties.
```

```yaml
Type: System.String[]
Parameter Sets: FoundSet
Aliases:

Required: False
Position: 0
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Scope

> Specify installer scope. Valid values are "Any", "User", "System", "UserOrUnknown", "SystemOrUnknown".

Reviewer 4: ```suggestion
Specify WinGet package installer scope. The parameter accepts the following values:

- `Any`
- `User`
- `System`
- `UserOrUnknown`
- `SystemOrUnknown`

> [!NOTE]
> The installer scope must be available in the WinGet package manifest.
```

```yaml
Type: Microsoft.WinGet.Client.PSObjects.PSPackageInstallScope
Parameter Sets: (All)
Aliases:
Accepted values: Any, User, System, UserOrUnknown, SystemOrUnknown

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -SkipDependencies

> Skip dependency installation.

Reviewer 4: ```suggestion
Specifies that the command shouldn't install the WinGet package dependencies.
```

```yaml
Type: System.Management.Automation.SwitchParameter
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Source

> Specify configured WinGet source.

Reviewer 4: ```suggestion
Specify the name of a configured WinGet source.
```

```yaml
Type: System.String
Parameter Sets: FoundSet
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Version

Specify package version.

```yaml
Type: System.String
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Confirm

Prompts you for confirmation before running the cmdlet.

```yaml
Type: System.Management.Automation.SwitchParameter
Parameter Sets: (All)
Aliases: cf

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -WhatIf

> Shows what would happen if the cmdlet runs.
> The cmdlet is not run.

Reviewer 4: ```suggestion
Shows what would happen if the cmdlet runs. The cmdlet isn't run.
```

 [-Architecture <PSProcessorArchitecture>] [-InstallerType <PSPackageInstallerType>] [-Locale <String>]
 [-Scope <PSPackageInstallScope>] [-SkipDependencies] [-Version <String>] [-Id <String>] [-Name <String>]
 [-Moniker <String>] [-Source <String>] [[-Query] <String[]>] [-MatchOption <PSPackageFieldMatchOption>]
 [-WhatIf] [-Confirm] [<CommonParameters>]
```

### GivenSet

```
Update-WinGetPackage [-IncludeUnknown] [-Mode <PSPackageInstallMode>] [-Override <String>] [-Custom <String>]
 [-Location <String>] [-Log <String>] [-Force] [-Header <String>] [-AllowHashMismatch]
 [-Architecture <PSProcessorArchitecture>] [-InstallerType <PSPackageInstallerType>] [-Locale <String>]
 [-Scope <PSPackageInstallScope>] [-SkipDependencies] [[-PSCatalogPackage] <PSCatalogPackage>]
 [-Version <String>] [-WhatIf] [-Confirm] [<CommonParameters>]
```

## DESCRIPTION

This command searches the packages installed on your system and installs a newer version of the
matching WinGet package. The command includes parameters to specify values used to search for
packages in the configured sources. By default, the command searches all sources. By default, all

> string-based searches are case-insensitive substring searches. Wildards are not supported. You can

Reviewer 4: ```suggestion
string-based searches are case-insensitive substring searches. Wildcards are not supported. You can
```

```

This example shows how to update a specific version of a package using a query. The command does a
query search for packages matching `Microsoft.PowerShell`. The results of the search a limited to
matches with the version of `7.4.4.0`.

## PARAMETERS

### -AllowHashMismatch

Allows you to download package even when the SHA256 hash for an installer or a dependency does not
match the SHA256 hash in the WinGet package manifest.

```yaml
Type: System.Management.Automation.SwitchParameter
Parameter Sets: (All)
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Architecture

Specify the processor architecture for the WinGet package installer. The parameter accepts the
following values:

- `Default`
- `X86`
- `Arm`
- `X64`
- `Arm64`

```yaml
Type: Microsoft.WinGet.Client.PSObjects.PSProcessorArchitecture
Parameter Sets: (All)
Aliases:
Accepted values: Default, X86, Arm, X64, Arm64

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByPropertyName)
Accept wildcard characters: False
```

### -Custom

> Use this paramater to pass additional arguments to the installer. The parameter takes a single

Reviewer 4: ```suggestion
Use this parameter to pass additional arguments to the installer. The parameter takes a single
```
